internal/exec/util/file.go

```diff
@@ -159,31 +161,66 @@ func (u Util) PerformFetch(f *FetchOp) error {
 		return err
 	}
 
-	// XXX(vc): Note that we assume to be operating on the file we just wrote, this is only guaranteed
-	// by using syscall.Fchown() and syscall.Fchmod()
+	if f.Append {
+		// Default to the appended file's owner for the uid and gid
+		defaultUid, defaultGid, mode := getFileOwnerAndMode(path)
+		uid, gid, err := u.GetNodeOwners(f.Node, defaultUid, defaultGid)
```

Comment:
If the node doesn't exist does `GetNodeOwners` return an error?

Reply:
It returns the `id` field or the result of a user name / group name lookup of the `name` field in the `types.Node` struct passed in, it doesn't look at the node on disk. It probably doesn't even exist yet at this point.